CMakeLists.txt

```diff
@@ -100,44 +89,19 @@ endif(DROGON_CXX_STANDARD LESS 17)
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake_modules/)
 
 # jsoncpp
-find_package(Jsoncpp REQUIRED)
-target_include_directories(${PROJECT_NAME} PUBLIC ${JSONCPP_INCLUDE_DIRS})
-target_link_libraries(${PROJECT_NAME} PUBLIC ${JSONCPP_LIBRARIES})
-list(APPEND INCLUDE_DIRS_FOR_DYNAMIC_VIEW ${JSONCPP_INCLUDE_DIRS})
-message(STATUS "jsoncpp inc path:" ${JSONCPP_INCLUDE_DIRS})
-
-# message(STATUS ${JSONCPP_LIBRARIES})
-if(NOT EXISTS ${JSONCPP_INCLUDE_DIRS}/json/version.h)
-  message(FATAL_ERROR "Error: jsoncpp lib is too old.....stop")
-endif()
+find_package(jsoncpp 1.7 REQUIRED)
```

Comment:
Here it seems the issue is that in earlier versions `jsoncpp` did not publish a version file:
See
https://github.com/open-source-parsers/jsoncpp/blob/1.7.0/CMakeLists.txt#L137 (old)
https://github.com/open-source-parsers/jsoncpp/blob/master/CMakeLists.txt#L217

Thus we may need to remove the `1.7` from the `find_package`. Note that in newer versions this documentation
https://cmake.org/cmake/help/latest/command/find_package.html#version-selection
implies that `JSONCPP_VERSION` or something like that will be defined. So perhaps we can fall back to the old version detecting logic only when that variable is not defined, else you can use the `VERSION_LESS` comparator in the `if` statement

That being said, your original approach has the advantage that it still works if the user is on a new version of `jsoncpp` but has disabled `JSONCPP_WITH_CMAKE_PACKAGE`. So in this case you may just wish to 
- keep the existing `findJsoncpp.cmake` and install it as we now do with `findUUID`; but
- don't do `target_include_directories(${JSONCPP_INCLUDE_DIR})`, instead do `add_library` to add jsoncpp as an `IMPORTED` library target and use `set_target_properties` to set its `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES`. Then you can do `target_link_libraries` to the imported target